roxctl/deployment/check/check.go

```diff
@@ -224,6 +250,18 @@ func (d *deploymentCheckCommand) getAlertsAndIgnoredObjectRefs(deploymentYaml st
 	for _, r := range response.GetRuns() {
 		alerts = append(alerts, r.GetAlerts()...)
 	}
+
+	// Deployment ID is empty because the processed yaml comes from roxctl and therefore doesn't
+	// get a  Kubernetes generated ID. This is a temporary ID only required for roxctl to distinguish
+	// between different generated deployments.
+	for _, alert := range alerts {
```

Comment:
Isn't it possible to have two entries in `alerts` that have the same entity (i.e. same deployment)? I believe this would cause every deployment entity to get assigned a new ID regardless of being the same entity. 

This could be related to the failing unit tests (did not check this): https://github.com/stackrox/stackrox/actions/runs/7035940956/job/19147480721?pr=8822

Reply:
I think you're right after some testing. I'll apply the same random ID to multiple deployments within the same alert then.